### PR TITLE
Fix WebDAV not found error handling

### LINKED ISSUE #906: "Error: Unable to unzip snapshot" if file is missing on WebDAV server?
https://forums.zotero.org/discussion/comment/461279/#Comment_461279

### PR DESCRIPTION
**Pull request by mvasilak**
Closes #906

### CODE REVIEW

File: Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift [hunk 1]
```diff
@@ -796,11 +796,27 @@ extension AttachmentDownloader: URLSessionDownloadDelegate {
             return
         }
 
+        var error: Swift.Error?
+        do {
```
Comment (michalrentka) [inline Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift:800]: No need to have this in `do catch` block, just have something like
```
let error = checkFileResponse(for: Files.file(from: location), fileStorage: fileStorage, downloadTask: downloadTask)

if let data = activeDownload.logData {
    logResponse(for: data, task: downloadTask, error: error)
}
...
if let error {
    ...
}
```

**Review by michalrentka — CHANGES_REQUESTED** [review]

File: Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift (hunk 1 above)
Comment (mvasilak) [reply Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift:800]: Correct, I just merged this, I'll just push another commit to improve this.

File: Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift [hunk 2]
```diff
@@ -848,11 +861,17 @@ extension AttachmentDownloader: URLSessionDownloadDelegate {
             }
         }
 
-        func checkFileResponse(for file: File, fileStorage: FileStorage) -> Swift.Error? {
+        func checkFileResponse(for file: File, fileStorage: FileStorage, downloadTask: URLSessionDownloadTask) -> Swift.Error? {
             let size = fileStorage.size(of: file)
-            if size == 0 || (size == 9 && (try? fileStorage.read(file)).flatMap({ String(data: $0, encoding: .utf8) }) == "Not found") {
+            if size == 0 || (size == 9 && (try? fileStorage.read(file)).flatMap({ String(data: $0, encoding: .utf8) })?.caseInsensitiveCompare("Not found") == .orderedSame) {
```
Comment (michalrentka) [inline Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift:866]: And while you're fixing the code above, you can extract this bit into a standalone function and use it both here and in `RemoteAttachmentDownloadOperation`

File: Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift (hunk 1 above)
Comment (mvasilak) [reply Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift:800]: Check https://github.com/zotero/zotero-ios/commit/919a1d40e913e3b4182d0bc76cc0bbf4bab870f6 (commited directly)

File: Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift (hunk 2 above)
Comment (mvasilak) [reply Zotero/Controllers/Attachment Downloader/AttachmentDownloader.swift:866]: Check https://github.com/zotero/zotero-ios/commit/919a1d40e913e3b4182d0bc76cc0bbf4bab870f6 (commited directly)

**Pull request merged**